src/QsCompiler/CompilationManager/CompilationUnitManager.cs

```diff
@@ -941,6 +942,23 @@ private void TypeCheckFile(string fileId, CancellationToken cancellationToken)
         public IEnumerable<Uri>? GetSourceFiles() =>
             this.FlushAndExecute(() => this.fileContentManagers.Keys.Select(id => new Uri(id)).ToImmutableArray().AsEnumerable());
 
+        /// <summary>
+        /// Gets whether a file is a notebook cell
+        /// </summary>
+        /// <returns>
+        /// A boxed bool holding true if it is a notebook cell, else false
+        /// </returns>
+        /// <remarks>
+        /// Waits for all currently running or queued tasks to finish
+        /// before getting the file content by calling <see cref="FlushAndExecute"/>.
+        /// </remarks>
+        public object? FileIsNotebookCell(TextDocumentIdentifier textDocument) =>
+
+            // Boxing needed here because FileQuery() is generic:
+            // https://devblogs.microsoft.com/dotnet/try-out-nullable-reference-types/#the-issue-with-t
+            this.FlushAndExecute(() =>
```

Comment:
I would recommend unboxing before returning to keep the public API clean. Alternatively, if you update the project to use C# 10, you can change `FlushAndExecute` and `FileQuery` to use `T?` which would allow you to use `bool` without boxing.

Reply:
I couldn't get it to cooperate on C# 10, but casting to `bool?` is less nasty, thanks